### DISCUSSION TITLE
Can I change java source directory?

### DISCUSSION THREAD

**Discussion by sxp** [davidB/scala-maven-plugin#533] [Q&A] [answered]
Can I change the Java source directory from 'src/main/java' to another?

**Comment by davidB** [comment]
Yes the same way, you can change for maven java compiler plugin. If you want to change the info only for this plugin, see configuration of [scala-maven-plugin – scala:compile](https://davidb.github.io/scala-maven-plugin/compile-mojo.html#source)

**Reply by sxp** [reply]
I set the source option, but the scala-maven-plugin still uses src/main/java as the Java source at the compile goal.

my config:

              <plugin>
                <groupId>net.alchim31.maven</groupId>
                <artifactId>scala-maven-plugin</artifactId>
                <executions>
                    <execution>
                        <id>scala-compile-first</id>
                        <phase>process-resources</phase>
                        <goals>
                            <goal>add-source</goal>
                            <goal>compile</goal>
                        </goals>
                        <configuration>
                            <source>target/generated-sources/delombok</source>
                        </configuration>
                    </execution>
                </executions>
            </plugin>

**Comment by sxp** [comment]
I read the configuration manual before, but can't find which option is to change the Java source directory.

sourceDir option is to change the scala source directory, and it is a File type, can't set to scala and Java source separately.

Which configuration should I use? And can you provide an example for use, thank you very much.

**Comment by sxp** [comment, marked as answer]
Finally, I set the `<build><sourceDirectory>` to change the Java source directory. It works.
